### PR TITLE
added validations for booking office and user

### PR DESCRIPTION
**Pull request by MG2022TK**

### CODE REVIEW

File: app/models/booking.rb [hunk 1]
```diff
@@ -1,4 +1,19 @@
 class Booking < ApplicationRecord
   belongs_to :user
   belongs_to :office
+  has_one :review, dependent: :destroy
+
+  validates :start_date, presence: true
+  validates :end_date, presence: true
+  validates :occupants, presence: true, numericality: { greater_than_or_equal_to: 0 }
```
Comment (margeol90) [inline app/models/booking.rb:8]: Actually this is something we left out from our schema!

File: app/models/office.rb [hunk 2]
```diff
@@ -1,3 +1,20 @@
 class Office < ApplicationRecord
+  has_one_attached :photo
   belongs_to :user
+  has_many :bookings, dependent: :destroy
+  has_many :reviews, through: :bookings
+  has_many :facilities
```
Comment (margeol90) [inline app/models/office.rb:6]: Facilities are also a TBC - we need to check with a TA.
But the validation can stay !

**Comment by margeol90** [comment]
@Jide-Muritala have you worked on this branch just now?

**Comment by Jide-Muritala** [comment]
Yes, I was just leaving a message for you

File: app/models/office.rb (hunk 2 above)
Comment (Jide-Muritala) [reply app/models/office.rb]: Yes, I have now removed facilities so that it doesn't cause errors since we have not yet created its migration.

**Comment by margeol90** [comment]
No problem! I wasn't sure what was going on because I left some comments for Michael.

**Pull request merged**